lib/using_restricted_array.rb

```diff
@@ -1,61 +1,106 @@
-require_relative 'restricted_array.rb'
+require_relative "restricted_array.rb"
 # RestrictedArray can be created using a specified size, or a random size in
 # the range of 1-20 will be chosen for you.
 # All values are integers in the range of 1-221.
 # RestrictedArray cannot be resized.
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n), where n is the number of items in the array or the length of the input array
+# Space complexity: O(1)
 def length(array)
-  raise NotImplementedError
+  i = 0
+
+  until array[i] == nil
+    i += 1
+  end
+
+  return i
 end
 
 # Prints each integer values in the array
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n), where n is the number of items in the array or the length of the input array
+# Space complexity: O(1)
 def print_array(array)
-  raise NotImplementedError
+  len = length(array)
+  i = 0
+
+  while i < len
+    print array[i]
+    print " "
+    put i
```

Comment:
Why is this line here? The problem specification just wants you to print out the array contents.